src/sql/azdata.d.ts

```diff
@@ -3281,16 +3281,17 @@ declare module 'azdata' {
 	}
 
 	export interface TextComponent extends Component, TextComponentProperties {
-		/**
-		 * An event called when the text is clicked
-		 */
-		onDidClick: vscode.Event<any>;
```

Comment:
Can you check this with Alan Ren? Any change that removes a public API should be verified. Did this _never_ work? 

I am concerned here about risk to break existing extensions - please at a minimum check Github and verify nothing is changed. I would strongly suggest holding on this removal, communicating via Github issue / verification with larger team and any partners like SQL Prompt that this isn't used, then and only then remove in an individual PR.

Reply:
this was introduced on Aug 27, It is very unlikely other folks are using this. but to be safe I like Kevin's idea and we can make it the process of updating/removing public APIs.

Reply:
It did work - I added it myself. I only added it for the BDC dashboard itself so it's highly unlikely anyone else is using it. The only reason I'm changing it is because it's not the right approach from an accessibility standpoint. 

The reason this breaks accessibility is because "clickable" is announced whenever an item has an onclick handler attached to it. And by hooking up the onclick in the angular component we were thus making EVERY text item "clickable" (even if nothing was being done).

A hyperlink is inherently a "clickable text" object so using that instead is the much better approach.

Reply:
We need to have a discussion about this - by leaving the existing click handler in on the text element we're not fixing the accessibility issue. How are we planning on handling updates to existing definitions (such as adding new properties/methods to classes)? That's going to be tricky to do "correctly" through the proposed API...

Reply:
Is there a way to annotate a property as deprecated so that anything building against it gets flagged?  Would be a good first step as we look to remove it.

Reply:
VS Code just uses the @deprecated tag

https://github.com/microsoft/azuredatastudio/blob/master/src/vs/vscode.d.ts#L1167

Give how few extensions are out there that aren't ours it's probably best to just manually check those to see if they're using the deprecated functions and if so reach out to the author directly. Figuring out some way to alert users of deprecated APIs would be a good idea long term though.